src/AutoMapper/Mappers/ExpressionMapper.cs

```diff
@@ -247,7 +247,13 @@ private Type GetSourceType(PropertyMap propertyMap)
             {
                 if (propertyMap.SourceMember == null)
                 {
-                    return propertyMap.SourceType;
+                    return propertyMap.SourceType ??
```

Comment:
If it's similar to XpressionMapper then either the custom expression or the source member should be available - otherwise how can the objects be mapped i.e. `var sourceType = propertyMap.CustomExpression == null
                    ? propertyMap.SourceMember.GetMemberType()
                    : propertyMap.CustomExpression.ReturnType;`

It does not seem like the code needs the source member except to find the type - may want to follow up with @TylerCarlson1